File: 파이썬/9226 도깨비말.py
def vowel(c):
  if c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u':
    return True
  else:
    return False


def dokkaebi(n):
  key = n
  while True:
    if vowel(n[0]):
      break

    n = n[1:] + n[0]
    if n == key:
      break

  n += "ay"
  return n


while True:
  s = input()

  if s == "#":
    break

  print(dokkaebi(s))

File: cpp/9226 도깨비말.cpp
#include <iostream>
using namespace std;

bool vowel(char c);
string dokkaebi(string n);

int main() {
  string s;

  while (true) {
    cin >> s;
    if (s == "#")
      break;

    cout << dokkaebi(s) << "\n";
  }
  
  return 0;
}

bool vowel(char c)
{
  switch (c) {
    case 'a': case 'e': case 'i': case 'o': case 'u':
      return true;
    default:
      return false;
  }
}

string dokkaebi(string n)
{
  string key = n;
  while (true) {
    if (vowel(n.at(0)))
      break;

    n += n.at(0);
    n.erase(0, 1);

    if (n == key) // 단어에 모음이 없어서 한 글자씩 뒤로 보내다가 원래 단어로 돌아온 경우
      break;
  }

  n += "ay";
  return n;
}
